rust/worker/src/blockstore/arrow/block/types.rs

```diff
@@ -92,6 +92,118 @@ impl Block {
         None
     }
 
+    pub fn get_prefix<'me, K: ArrowReadableKey<'me>, V: ArrowReadableValue<'me>>(
+        &'me self,
+        prefix: &str,
+    ) -> Option<Vec<(&str, K, V)>> {
+        let prefix_array = self
+            .data
+            .column(0)
+            .as_any()
+            .downcast_ref::<StringArray>()
+            .unwrap();
+        let mut res: Vec<(&str, K, V)> = vec![];
+        for i in 0..self.data.num_rows() {
+            let curr_prefix = prefix_array.value(i);
+            if curr_prefix == prefix {
+                res.push((
+                    curr_prefix,
+                    K::get(self.data.column(1), i),
+                    V::get(self.data.column(2), i),
+                ));
+            }
+        }
+        return Some(res);
+    }
+
+    pub fn get_gt<'me, K: ArrowReadableKey<'me>, V: ArrowReadableValue<'me>>(
+        &'me self,
+        prefix: &str,
+        key: K,
+    ) -> Option<Vec<(&str, K, V)>> {
+        let prefix_array = self
+            .data
+            .column(0)
+            .as_any()
+            .downcast_ref::<StringArray>()
+            .unwrap();
+        let mut res: Vec<(&str, K, V)> = vec![];
+        for i in 0..self.data.num_rows() {
+            let curr_prefix = prefix_array.value(i);
+            let curr_key = K::get(self.data.column(1), i);
+            if curr_prefix > prefix || (curr_prefix == prefix && curr_key > key) {
```

Comment:
I think we want to compare prefix equality here and only compare the inequality on keys. @HammadB ?

Reply:
As an example, in metadata indices the prefix is the metadata key, the blockfile key is the metadata value, and the blockfile value is a roaring bitmap of offset IDs. We want strict prefix equality there, not an inequality comparison

Reply:
Yes